bundler/lib/bundler/ruby_dsl.rb

```diff
@@ -11,7 +11,10 @@ def ruby(*ruby_version)
 
       if options[:file]
         raise GemfileError, "Cannot specify version when using the file option" if ruby_version.any?
-        ruby_version << Bundler.read_file(Bundler.root.join(options[:file])).strip
+        file_content = Bundler.read_file(Bundler.root.join(options[:file]))
+        matched_ruby_version = /^ruby\s+([\d.]+)(?:\s*#\s*.*)?$/.match(file_content)
+        # if a line in the file begins with "ruby" use that else use the whole file
+        ruby_version << matched_ruby_version ? matched_ruby_version[1].strip : file_content.strip
```

Comment:
I was thinking about how to handle comments too. I think it might be better to take the whole line, then split at `#` and use whatever comes before it. Here's an approach that hopefully isn't _too_ clever.

```suggestion
        if /^ruby\s+(.*)$/.match(file_content)
          ruby_version << $1.split("#", 2).first.strip
        else
          ruby_version << file_content.strip
        end
```

 Trying too hard to match the version number is going to cause problems since ruby versions can be specified in ways that aren't just standard version numbers.

Reply:
I really like this solution, extremely reasonable and I would say nice and simple. Should work great and we don't need tons of code here for this simple task.